### PR TITLE
Copter/Scripting: add get_circle_radius and set_circle_rate

### PR DESCRIPTION
**Pull request by rmackay9**
This allows scripts to read the radius and write the rotation rate of a copter in Circle mode (or executing LOITER_TURNS).  This has been tested in SITL and resolves a 4.1 issue reported by Arace.

### CODE REVIEW

**Comment by IamPete1** [comment]
I think I would prefer to see this done as a direct singleton to `AC_Circle` so we don't have to clutter AP_Vehicle with bindings that are only used one one vehicle. It also means we can `#define` out the whole singleton on vehicles that are not copter, we don't currently support per-method defines. A sort of precedent for this is the quadplane bindings that go direct with a specially added quadplane singleton.

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 1]
```diff
@@ -206,6 +206,8 @@ singleton AP_Vehicle method get_control_output boolean AP_Vehicle::ControlOutput
 singleton AP_Vehicle method get_target_location boolean Location'Null
 singleton AP_Vehicle method set_target_velocity_NED boolean Vector3f
 singleton AP_Vehicle method set_target_angle_and_climbrate boolean float -180 180 float -90 90 float -360 360 float -FLT_MAX FLT_MAX boolean float -FLT_MAX FLT_MAX
+singleton AP_Vehicle method get_circle_radius boolean float'Null
+singleton AP_Vehicle method set_circle_rate boolean float -90 90
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:210]: I have used up 150 deg/s https://youtu.be/m5bwoiHsn8M?t=10

+- 360 should be enough, but I would be quite tempted not to check at all, the vehicle will still be limited by `WP_NAV_SPEED`

Comment (rmackay9) [reply libraries/AP_Scripting/generator/description/bindings.desc]: ok, I've removed the limit on the rate that can be set.

**Comment by rmackay9** [comment]
@IamPete1, thanks very much for the review.  I also thought about a direct binding to AC_Circle but after some thought I don't think it's a good idea.  The reason is just that it makes it vehicle specific but also more importantly I think it is a layering violation and is inconsistent with how we control the vehicle in Guided mode.

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 2]
```diff
@@ -206,6 +206,8 @@ singleton AP_Vehicle method get_control_output boolean AP_Vehicle::ControlOutput
 singleton AP_Vehicle method get_target_location boolean Location'Null
 singleton AP_Vehicle method set_target_velocity_NED boolean Vector3f
 singleton AP_Vehicle method set_target_angle_and_climbrate boolean float -180 180 float -90 90 float -360 360 float -FLT_MAX FLT_MAX boolean float -FLT_MAX FLT_MAX
+singleton AP_Vehicle method get_circle_radius boolean float'Null
+singleton AP_Vehicle method set_circle_rate boolean float -FLT_MAX FLT_MAX
```
Comment (peterbarker) [inline libraries/AP_Scripting/generator/description/bindings.desc:210]: Don't we save bytes by omitting the limits now?

Comment (rmackay9) [reply libraries/AP_Scripting/generator/description/bindings.desc:210]: I tried removing the limit but I received an error message regarding the lower limit missing..

Comment (IamPete1) [reply libraries/AP_Scripting/generator/description/bindings.desc:210]: the no check stuff is not in yet, https://github.com/ArduPilot/ardupilot/pull/18221

**Review by tridge — APPROVED** [review]

**Pull request merged**